add docs for new cl columns: (#105)
cleanlab/cleanlab-studio: docs/source/concepts/cl_columns.rst

>    * - Cleanlab Column Name

Reviewer: Can sort/group these by what they apply to, e.g., group label-related ones together

     - Value Type
   * - :ref:`is_label_issue <is_label_issue>`
     - Boolean
   * - :ref:`label_issue_score <label_issue_score>`
     - Float
   * - :ref:`suggested_label <suggested_label>`
     - String
   * - :ref:`is_outlier <is_outlier>`
     - Boolean
   * - :ref:`outlier_score <outlier_score>`
     - Float
   * - :ref:`is_ambiguous <is_ambiguous>`
     - Boolean
   * - :ref:`ambiguous_score <ambiguous_score>`
     - Float
   * - :ref:`is_given_label_correct <is_given_label_correct>`
     - Boolean
   * - :ref:`is_near_duplicate <is_near_duplicate>`
     - Boolean
   * - :ref:`near_duplicate_score <near_duplicate_score>`
     - Float
   * - :ref:`near_duplicate_id <near_duplicate_id>`

>      - List of integers

Reviewer: not a list of integers anymore

Label Issues
============

.. _is_label_issue:
``is_label_issue``
-----
Contains a boolean value, with ``True`` indicating that the sample is likely to have a label issue. The value is obtained by thresholding the ``label_issue_score`` score with confident learning.

.. _label_issue_score:
``label_issue_score``
-------------
Contains a score bounded between 0 and 1. The score is calculated using confident learning. The lower the score of a sample, the more likely it has a label issue.

.. _suggested_label:
``suggested_label``
---------------
Contains the suggested label for the sample. If the sample is not a label issue (``is_label_issue`` marked as ``False``), the suggested label will be empty. For samples with label issues, the suggested label is computed by Cleanlab studio.

Outliers
========

.. _is_outlier:
``is_outlier``
-------

> Contains a boolean value, with ``True`` indicating that the sample is likely to be an outlier.

Reviewer 2: ```suggestion
Contains a boolean value, with ``True`` indicating that the sample is likely to be an outlier. 
Note that for tabular-mode multi-label projects, is_outlier will always return False. 
```

Author: already merged

     - Boolean
   * - :ref:`is_near_duplicate <is_near_duplicate>`
     - Boolean
   * - :ref:`near_duplicate_score <near_duplicate_score>`
     - Float
   * - :ref:`near_duplicate_id <near_duplicate_id>`
     - Integer

Label Issues
============

.. _is_label_issue:
``is_label_issue``
-----
Contains a boolean value, with ``True`` indicating that the sample is likely to have a label issue. The value is obtained by thresholding the ``label_issue_score`` score with confident learning.

.. _label_issue_score:
``label_issue_score``
-------------
Contains a score bounded between 0 and 1. The score is calculated using confident learning. The lower the score of a sample, the more likely it has a label issue.

.. _suggested_label:
``suggested_label``
---------------
Contains the suggested label for the sample. If the sample is not a label issue (``is_label_issue`` marked as ``False``), the suggested label will be empty. For samples with label issues, the suggested label is computed by Cleanlab studio.

Outliers
========

.. _is_outlier:
``is_outlier``
-------
Contains a boolean value, with ``True`` indicating that the sample is likely to be an outlier.

> Note: for projects on multi-label tabular datasets, outliers are currently not computed, and ``is_outlier`` will always be False. 

Reviewer: This note renders as gigantic text

<img width="841" alt="Screenshot 2023-06-27 at 3 21 29 PM" src="https://github.com/cleanlab/cleanlab-studio/assets/3526486/ea9fd175-8c94-45d4-9ad3-daed6837e30f">

``ambiguous_score``
-------------
Contains a score bounded between 0 and 1, which is used to determine whether a sample is ambiguous. The lower the score of a sample, the more likely it is to be ambiguous.

High Confidence
===============

.. _is_high_confidence_given_label:
``is_high_confidence_given_label``
---------------
Contains a boolean value, with ``True`` indicating that the given label of the sample is likely to be correct, so the sample can be safely used in downstream tasks.

Near Duplicates
===============

.. _is_near_duplicate:
``is_near_duplicate``
----------------
Contains a boolean value, with ``True`` indicating that the sample is likely to be a near duplicate of another sample. Near duplicates are two or more examples in a dataset that are extremely similar (or identical) to each other, relative to the rest of the dataset.

.. _near_duplicate_score:
``near_duplicate_score``
------------------
Contains a score bounded between 0 and 1, which is used to determine whether a sample is a near duplicate. The lower the score of a sample, the more likely it is to be a near duplicate of another sample.

.. _near_duplicate_id:
``near_duplicate_id``
----------------

> Contains an integer ID for each sample, where samples with the same ID are near duplicates of each other. The IDs range from 0 upwards. Samples that do not have near duplicates are assigned an ID of -1.

Reviewer: Remember to update this text once we switch the representation to use NA instead of -1.

     - String

>    * - :ref:`is_outlier <is_outlier>`

Author: for outliers specifically, when we find outliers in studio with pred_probs (for tabular data), we don't just threshold a single score but instead use the whole pred_probs matrix and do some per-class thresholding, so there is no unified score

For ambiguous there is a score, I'll add it

>    * - :ref:`is_outlier <is_outlier>`

Author: we added scores now

     - Boolean
   * - :ref:`is_ambiguous <is_ambiguous>`
     - Boolean
   * - :ref:`is_high_confidence <is_high_confidence>`
     - Boolean
   * - :ref:`is_uncertain <is_uncertain>`
     - Boolean
   * - :ref:`is_near_duplicate <is_near_duplicate>`
     - Boolean
   * - :ref:`near_duplicate_score <near_duplicate_score>`
     - Float

>    * - :ref:`near_duplicate_sets <near_duplicate_sets>`

Reviewer: And if we do stick with list-of-integers: will the Pandas DataFrame contain a Python list of integers?

.. _suggested_label:
``suggested_label``
---------------
Contains the suggested label for the sample. If the sample is not a label issue (``is_label_issue`` marked as ``False``), the suggested label will be empty. For samples with label issues, the suggested label is computed by Cleanlab studio.

Outliers
========

.. _is_outlier:
``is_outlier``
-------
Contains a boolean value, with ``True`` indicating that the sample is likely to be an outlier.

Ambiguous
=========

.. _is_ambiguous:
``is_ambiguous``
----------
Contains a boolean value, with ``True`` indicating that the sample is likely to be ambiguous. Ambiguous samples are those that do not obviously belong to a single class.

High Confidence
===============

.. _is_high_confidence:
``is_high_confidence``
---------------
Contains a boolean value, with ``True`` indicating high confidence, which means that the given label of the sample is likely to be correct, so the sample can be safely used in downstream tasks.

Uncertain
=========

.. _is_uncertain:
``is_uncertain``
-----------

> Contains a boolean value, with ``True`` indicating uncertain. Uncertain data specifically means data that are likely to be mislabeled (they will have ``is_label_issue`` marked ``True``), but that Cleanlab is uncertain about what their correct label should be. Such data would benefit from human review.

Reviewer: Is this just the negation of `is_high_confidence`? If yes, remove it; if not, would be nice to briefly explain how the two are different.

Author: they are related but not quite negations:
- `is_high_confidence` (which will become `is_given_label_correct` is where P[suggested] is high and label quality is high
- `is_uncertain` is where sample is marked with label issue, and P[suggested] is low

But as per discussion we will not include `is_uncertain` for now because of its similarity with ambiguous (see https://cleanlabinc.slack.com/archives/C02SN6TFSF8/p1686948331146719)

Near Duplicates
===============

.. _is_near_duplicate:
``is_near_duplicate``
----------------

> Contains a boolean value, with ``True`` indicating that the sample is likely to be a near duplicate of another sample. Near duplicates are two or more examples in a dataset that are extremely similar to each other, relative to the rest of the dataset. The examples flagged with this issue may be exactly duplicated, or lie atypically close together when represented as vectors (i.e. feature embeddings).

Reviewer: ```suggestion
Contains a boolean value, with ``True`` indicating that the sample is likely to be a near duplicate of another sample. Near duplicates are two or more examples in a dataset that are extremely similar (or identical) to each other, relative to the rest of the dataset.
```

Suggestion, omit the implementation detail that we're looking for data points that are close in embedding space. The other explanations don't include such detail.
